include/service/entities/site_service.js

```diff
@@ -0,0 +1,491 @@
+    SiteService.SITE_FIELD = 'site';
+    SiteService.SITE_COLLECTION = 'site';
+    var SITE_COLL = SiteService.SITE_COLLECTION;
+
+
+    /**
+     * Load full site config from the database using the unique id.
+     * @method getByUid
+     * @param {String} uid - unique id of site
+     * @param {Function} cb - the callback function
+     */
+    SiteService.prototype.getByUid = function(uid, cb) {
+        if(!uid || uid === SiteService.GLOBAL_SITE) {
+            cb(null, {
+                displayName:pb.config.siteName,
+                hostname: pb.config.siteRoot,
+                uid: SiteService.GLOBAL_SITE
+            });
+        }
+        else {
+            var dao = new pb.DAO();
+            var where = {uid: uid};
+            dao.loadByValues(where, SITE_COLL, cb);
+        }
+    };
+
+    /**
+     * Get all of the site objects in the database
+     * @method getAllSites
+     * @param {Function} cb - the callback function
+     */
+    SiteService.prototype.getAllSites = function(cb) {
+        var dao = new pb.DAO();
+        dao.q(SITE_COLL, { select: pb.DAO.SELECT_ALL, where: {} }, cb);
+    };
+
+    /**
+     * Get all site objects where activated is true.
+     * @method getActiveSites
+     * @param {Function} cb - the callback function
+     */
+    SiteService.prototype.getActiveSites = function(cb) {
+        var dao = new pb.DAO();
+        dao.q(SITE_COLL, { select: pb.DAO.SELECT_ALL, where: {active: true} }, cb);
+    };
+
+    /**
+     * Get all site objects where activated is false.
+     * @method getInactiveSites
+     * @param {Function} cb - the callback function
+     */
+    SiteService.prototype.getInactiveSites = function(cb) {
+        var dao = new pb.DAO();
+        dao.q(SITE_COLL, {where: {active: false}}, cb);
+    };
+
+    /**
+     * Get all site objects segmented by active status.
+     * @method getSiteMap
+     * @param {Function} cb - the callback function
+     */
+    SiteService.prototype.getSiteMap = function(cb) {
+        var self  = this;
+        var tasks = {
+
+             active: function(callback) {
+                 self.getActiveSites(callback);
+             },
+
+             inactive: function(callback) {
+                 self.getInactiveSites(callback);
+             }
+        };
+        async.series(tasks, function(err, results) {
+            cb(err, results);
+        });
+    };
+
+    /**
+     * Get site name given a unique id.
+     * @method getSiteNameByUid
+     * @param {String} uid - unique id
+     * @param {Function} cb - the callback function
+     */
+    SiteService.prototype.getSiteNameByUid = function(uid, cb) {
+        var dao = new pb.DAO();
+        dao.q(SITE_COLL, {select: pb.DAO.SELECT_ALL, where: {uid: uid} }, function(err, result) {
+            var siteName = (!uid || uid === SiteService.GLOBAL_SITE) ? 'global' : '';
+
+            if(pb.util.isError(err)) {
+                pb.log.error(err);
```

Comment:
Why do we silently fail here instead of reporting the error back through the callback?